geopandas/sindex.py

```diff
@@ -81,22 +223,119 @@ def is_empty(self):
 
 if compat.HAS_PYGEOS:
 
-    from pygeos import STRtree, box, points  # noqa
+    import geopandas
```

Comment:
Just a quick note, (I'll do a proper look later), can you import only classes you need from their files instead of whole geopandas?

Reply:
I was running into circular imports with `geopandas.geoseries.GeoSeries`. We can either import it as `import geopandas.geoseries` here or I think we could also change `from .sindex import get_sindex_class, has_sindex` to `import sindex` in `geopandas.base` instead which would allow `from geopandas import GeoSeries`.